index.rst

```diff
@@ -48,10 +48,205 @@
+
+  import numpy as np
+  import lsst.efd as efd
+  ...
+
+  # Get the temperatures in one go 
+  envtemp = efd.get("m1m3.actuators.envtemp")
+  stdev = np.std(envtemp)
+  mean = np.mean(envtemp)
+  print(f"temperature = {mean} +/- {stdev} K")
+
+
+
+Aggregation
+-----------
+
+The purpose of aggregation is both to reduce volume on high-frequency telemetry data and to increase the signal-to-noise of busy telemetry. Science Platform users are generally interested in events at the same order of cadence as a camera exposure; therefore we propose that all telemetry data sampled with a frequency higher than 1Hz is (1) sampled at 1Hz and (2) aggregated to 1Hz using these generic statistics:
+
+* Max
+
+* Min
+
+* Mean
+
+* Median
+
+* Standard Deviation
+
+For command streams, no aggregation should be done.
+
+For event streams we propose that using Kafka we sum repeats of the
+same messages within the 1Hz window (eg if the M1M3 subsystem issues a
+limitError even 100 times in the last second, a repeat counter of 100
+is stored with the event).
+
+
+Latency
+-------
+
+Latency should be addressed in two parts:
+
+1. Persistence latency -- This is the latency between an even being published on the DDS to that event showing up as an aggregated quantity in the DM EFD.  This latency should be equal to or less than the time to take and reduce a single raft of data on a parallel reduction system.  This puts an upper bound on the sampling rate for the aggregated event streams. For Auxillary data, lower latencies are required; for example CBP data has been requested to be available at 1-second scale latencies. 
+
+2. Query latency -- Doing a strict time span query should be of order 1 sec.  More complicated queries, queries involving joins, will have higher latency and should be addressed on a case by case basis.
+
+Redundancy
+----------
+
+DM-EFD should be sized to hold the aggregated event streams from commissioning to the end if operations.  It should be redundant, or backed up so that the risk of data loss is acceptably low, even if the EFD system is backed some other way into cold storage. 
+
+Other
+-----
+
+A notebook examining data should be deployment invariant within LSST operations; i.e. the same notebook should work in a Science Platform deployment at the LDF and one at the Base. 
+
+Units should be SI units, and the time stamps should be in UTC.
+
+Example use cases
+=================
+
+The Science Platform is intended to support free-form troubleshooting
+for the Commissioning and Science Validation teams, so obtaining an
+exhaustive list of usecases is unlikely, especially given the time frame. As long as data is exposed in good time and in good form to the Science Platform, users will be happy.
```

Comment:
Saying that "if we do this, users will be happy" is insufficient; in particular, it doesn't address "if we do this other thing that costs less, users will also be happy" or "if we did this, we couldn't do that".

Reply:
This is not meant to be a requirements document or a trade study.  This is a vision document to put forward a design that we think is reasonable and addresses the obvious concerns of the end users.  Obviously, we have to do that trade study and a detailed design, but I don't think that is the job of this document.